app/assets/javascripts/admin/controllers/admin-web-hooks-show.js.es6

```diff
@@ -80,12 +80,24 @@ export default Ember.Controller.extend({
   actions: {
     save() {
       this.set('saved', false);
-
+      const url = extractDomainFromUrl(this.get('model.payload_url'));
       const model = this.get('model');
-      return model.save().then(() => {
-        this.set('saved', true);
-        this.get('controllers.adminWebHooks').get('model').addObject(model);
-      }).catch(popupAjaxError);
+      const saveWebHook = () => {
+        return model.save().then(() => {
+          this.set('saved', true);
+          this.get('controllers.adminWebHooks').get('model').addObject(model);
+        }).catch(popupAjaxError);
+      };
+
+      if (url.indexOf('localhost') !== -1 || url.match(/192\.168\.\d+\.\d+/) || url.match(/127\.\d+\.\d+\.\d+/) || url === Discourse.BaseUrl) {
```

Comment:
Sure. The check is all the way before validation so the url is starting with scheme. `extractDomainFromUrl` is the helper. It gets rid of path, scheme and port.

It yields:

- `https://meta.discourse.org:90/random` -> `meta.discourse.org`: domain name
- `http://192.168.0.1:80/random` -> `192.168.0.1`
- `http://localhost:21/` -> `localhost`

Then I performed `===` for both possible local url.